Alerts without a "message" annotation don't show meaningful details in MetalK8s UI

n1603 opened the pull request:
**Component**:

ui, tests

**Context**: 
https://github.com/scality/metalk8s/issues/2994

**Summary**:
Changes made to display alert message on UI when prometheus alerts annotation have any of description, summary or message field. Currently blank alert message shown for many of prometheus alerts in metalk8s UI.

**Acceptance criteria**: 


---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #ISSUE_NUMBER

<!-- If you want to refer to an issue while not closing it, use:

See: #ISSUE_NUMBER

-->

n1603 committed:
ui: Expand support Prometheus alerts message display

Changes made to display alert message on UI when prometheus alerts annotation have any of description, summary or message field

Refs: #2994

Signed-off-by: Ganesh Chaudhari <ganesh0407@gmail.com>

n1603 committed:
ui: Expand support Prometheus alerts message display

Changes made to display alert message on UI when prometheus alerts annotation have any of description, summary or message field

Refs: #2994

Signed-off-by: Ganesh Chaudhari <ganesh0407@gmail.com>

n1603 committed:
tests: Update monitoring test for alert

Change made to add additional field for Prometheus alerts with any of description, summary or message field

Refs: #2994

Signed-off-by: Ganesh Chaudhari <ganesh0407@gmail.com>

n1603 committed:
tests: Update monitoring test for alert

Change made to add additional field for Prometheus alerts with any of description, summary or message field

Refs: #2994

Signed-off-by: Ganesh Chaudhari <ganesh0407@gmail.com>

bert-e commented:
# Hello n1603,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

n1603 committed:
ui: Expand support Prometheus alerts message display

removed un-relaed change

Refs: #2994

Signed-off-by: Ganesh Chaudhari <ganesh0407@gmail.com>

ChengYanJin reviewed (commented):
Hello @n1603 ,

Thanks for creating the pull request!

The UI part looks good to me. 

If you can squash the first and third commit into one would be great!

JBWatenbergScality commented on ui/src/components/VolumeAlertsTab.js:
```diff
@@ -97,12 +97,21 @@ const ActiveAlertsCard = (props) => {
+       alert_description: alert_description,
+       active_since: alert.startsAt,
+     };
+     return alertData;
```
Thanks for your contribution.
You might want to simplify this to :
```js
return {
        name: alert.labels.alertname,
        severity: alert.labels.severity,
        alert_description: alert.annotations.description || alert.annotations.summary || alert.annotations.message,
        active_since: alert.startsAt,
      }
```

n1603 replied:
sure, will do that.

JBWatenbergScality commented on ui/src/containers/ClusterMonitoring.js:
```diff
@@ -169,12 +169,21 @@ const ClusterMonitoring = (props) => {
+        message: message,
         activeAt: alert.startsAt,
       };
+      return alertData;
```
Similarly this could be simplified as described before.

n1603 replied:
ok, will do. Thanks.

n1603 replied:
resolve in this pR : https://github.com/scality/metalk8s/pull/3030

n1603 committed:
ui: Expand support Prometheus alerts message display

Changes made to display alert message on UI when prometheus alerts annotation have any of description, summary or message field

Refs: #2994

Signed-off-by: Ganesh Chaudhari <ganesh0407@gmail.com>

n1603 committed:
Merge branch 'development/2.7' of https://github.com/n1603/metalk8s into development/2.7

n1603 commented:
> Hello @n1603 ,
> 
> Thanks for creating the pull request!
> 
> The UI part looks good to me.
> 
> If you can squash the first and third commit into one would be great!

closed this PR, as squashing did not work as expected. created : https://github.com/scality/metalk8s/pull/3030
